src/lower/mod.rs

```diff
@@ -909,24 +910,42 @@ impl Anonymize for [ir::ParameterKind<ir::Identifier>] {
 }
 
 impl ir::StructDatum {
-    fn to_program_clauses(&self) -> Vec<ir::ProgramClause> {
+    fn to_program_clauses(&self, program: &ir::Program) -> Vec<ir::ProgramClause> {
         // Given:
         //
-        //    struct Foo<T: Eq> { ... }
+        //    struct Foo<T: Eq> {
+        //        field: Bar
+        //    }
         //
         // we generate the following clause:
         //
-        //    for<?T> WF(Foo<?T>) :- (?T: Eq).
+        //    for<?T> WF(Foo<?T>) :- WF(?T), WF(Bar), (?T: Eq), WF(?T: Eq).
```

Comment:
We would never use `StructDefnWF`. Rather, `StructDefnWF` would be one of the things we must prove to decide if the whole program type-checks. This is basically a modularity thing -- it's the same as saying that if you have some function `foo`, we just need its *signature* to type-check the callers, but then we separately type-check the body of `foo` against its signature. 

Here, the idea is that, for some struct S, the users of S need not know the types of its fields; they can assume that if the type S is well-formed, then the types of its fields are well-formed.